FpsCheck.py

```diff
@@ -409,51 +409,47 @@ def find_invalid_alias_eSLDs(self, check_sets):
-            if curr_set.ccTLDs:
-                for aliased_site in curr_set.ccTLDs:
-                    # first check if the aliased site is actually anywhere else
-                    # in the fps
-                    if aliased_site != primary:
-                        if curr_set.associated_sites:
-                            sites = curr_set.associated_sites 
-                        else:
-                            sites = []
-                        if curr_set.service_sites:
-                            sites += curr_set.service_sites
-                        if aliased_site not in sites:
-                            self.error_list.append(
-                                "The aliased site " + aliased_site + 
-                                " contained within the ccTLDs must be a " +
-                                "primary, associated site, or service site " +
-                                "within the firsty pary set for " + primary)
-                    # check the validity of the aliases
-                    aliased_domain, aliased_tld = aliased_site.split(".", 1)
-                    if aliased_tld in self.icanns:
-                        icann_check = self.icanns.union({"com"})
-                    else:
-                        icann_check = self.icanns
-                    plus1s = [(site, site.split(".")[0], site.split(".")[-1])
-                              for site in curr_set.ccTLDs[aliased_site]]
-                    for eSLD in plus1s:
-                        if eSLD[1] != aliased_domain:
-                            self.error_list.append(
-                                "The following top level domain must match: " 
-                                + aliased_site + ", but is instead: " 
-                                + eSLD[0])
-                        if eSLD[2] not in icann_check:
-                            self.error_list.append(
-                                "The provided country code: " + eSLD[2] + 
-                                ", in: " + eSLD[0] + 
-                                " is not a ICANN registered country code")
+        for primary, curr_set in check_sets.items():
+            if not curr_set.ccTLDs:
+                continue
+            for aliased_site in curr_set.ccTLDs:
+                # first check if the aliased site is actually anywhere else
+                # in the fps
+                if not curr_set.includes(aliased_site, False):
+                    self.error_list.append(
+                        "The aliased site " + aliased_site + 
+                        " contained within the ccTLDs must be a " +
+                        "primary, associated site, or service site " +
+                        "within the firsty pary set for " + primary)
+                # check the validity of the aliases
+                aliased_domain, aliased_etld = (aliased_site.split(".")[0],
+                                                aliased_site.split(".")[-1])
```

Comment:
Taking another look at this for the eTLD/TLD change - I'm not sure that this is totally correct. Take for example, the eTLD+1 `cfredric.github.io` (note that `github.io` is on the PSL, so this is really an eTLD+1). For that site, `aliased_site` would be `"https://cfredric.github.io"`, so `aliased_domain` and `aliased_etld` work out to be `"cfredric"` and `"io"`, respectively. But the eTLD should be `"github.io"`, not just `"io"`.

We could adjust the indexing done here in order to get this case right, but it would be thrown off again if I specified the "site" as `"https://www.cfredric.github.io"`. (I'm not sure if we have a check for this kind of input; `FpsCheck.is_eTLD_Plus1` would not catch this.)

I think we need to use the `publicsuffix2` library to really be sure we're using the correct eTLD (up to changes between PSL snapshots, anyway).

Reply:
I'm not sure I'm fully understanding the issue. This check is meant to ensure that the TLD for a ccTLD variant is actually a country-code, so in the case of `https://cfredric.github.io` all we care is that `io` is a country code recognized by ICANN and therefore `https://cfredric.github.io` is considered a valid variant for another site such as `https://cfredric.com`. As far as I'm aware, this is intended behavior. Should it not be? 

However, I do agree that the `www.` subdomain will throw off how we check the eSLDs, which is something we can try to figure out.

Reply:
Resolved offline, the description/title of the PR got the intent backwards, so I misunderstood.